pkg/api/types.go

```diff
@@ -2793,6 +2793,11 @@ type ExportOptions struct {
 	Export bool `json:"export"`
 	// Should the export be exact.  Exact export maintains cluster-specific fields like 'Namespace'
 	Exact bool `json:"exact"`
+	// If specified:
+	// - if unset, then the result result is returned from remote storage based on quorum-read flag;
+	// - if it's 0, then we simply return what we currently have in cache, no guarantee;
+	// - if set to non zero, then the result is as fresh as given rv.
+	ResourceVersion string
```

Comment:
Brief summary of thinking from the meta.k8s.io issue:

1. GET, LIST, and WATCH are about returning a representation of a resource (default is the object in the schema corresponding to the kind in JSON)
2. Export is a special representation of the resource
3. Getting other alternate representations (as generic object meta, or as a tabular display) is higher level than GetOptions, and ExportOptions is more specialized than GetOptions (export options apply to one particular alternate representation).

So no matter what, for clients to be able to ask for v1.Pod in another form will need to pass additional modifiers - any values that apply to all of them should be in GetOptions.  Right now that is `resourceVersion` and `timeout`.

So if we want to create a client GetOptions that is passed to Get(), I'm ok with it.  However, we will almost certainly have to handle some of the other edge cases (that can't be represented with GetOptions) in the future, which means two refactors to clients (disruptive to clients).